File: work/DSA/c++/recursion_1b/stringToInteger.cpp
#include <bits/stdc++.h>
using namespace std;

int stringToNumber(char input[])
{
    if (input[0] == '\0')
    {
        return 0;
    }
    int smallAnswer = stringToNumber(input + 1);
    return smallAnswer + (input[0] - 48) * pow(10, strlen(input) - 1);
}

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);

    char input[50];
    cin >> input;
    cout << stringToNumber(input) << endl;
}

File: work/DSA/javascript/recursion_1b/stringToInteger.js
let input = ["0", "0", "0", "1", "2", "3", "4"]

function stringToNumber(input) {
    if (input[0] === undefined) {
        return 0;
    }
    let smallAnswer = stringToNuber(input.slice(1))
    return smallAnswer + (10 ** (input.length - 1)) * input[0]

}

let answer = stringToNumber(input)
console.log(answer)